run3/flow/ML/ComputeV2vsFDFrac.py

```diff
@@ -0,0 +1,323 @@
+(
+    hV2,
+    hNpFrac,
+    hFracvsV2,
+    cFrac,
+    hNpV2vsPt,
+    hPrV2vsPt,
+    hGraphV2,
+    hPrFrac,
+    AverageV2,
+    FinalErroXl,
+    FinalErroXh,
+) = ([], [], [], [], [], [], [], [], [], [], [])
+
+for (
+    inFileNameV2,
+    inFileNameFrac,
+) in zip(inputFilesV2, inputFilesFrac):
+    inFileNameV2 = os.path.join(inputDirV2, inFileNameV2)
+    inFileV2 = TFile.Open(inFileNameV2)
+    hV2.append(inFileV2.Get(histoNameV2))
+    AverageV2.append(inFileV2.Get(graphNameV2))
+    hV2[-1].SetDirectory(0)
+
+    inFileNameFrac = os.path.join(inputDirFrac, inFileNameFrac)
+    inFileFrac = TFile.Open(inFileNameFrac)
+    hNpFrac.append(inFileFrac.Get(histoNameFracFD))
+    hNpFrac[-1].SetDirectory(0)
+    hPrFrac.append(inFileFrac.Get(histoNameFracPr))
+    hPrFrac[-1].SetDirectory(0)
+
+SetGlobalStyle(
+    padbottommargin=0.13,
+    padleftmargin=0.12,
+    titleoffsety=1.0,
+    titlesize=0.05,
+    labelsize=0.05,
+)
+
+legDistr = TLegend(0.45, 0.69, 0.75, 0.89)
+legDistr.SetFillStyle(0)
+legDistr.SetBorderSize(0)
+legDistr.SetTextSize(0.045)
+
+cNonPromptV2 = TCanvas("cNonPromptV2", "non-prompt v2 versus pt")
+cNonPromptV2.SetCanvasSize(800, 800)
+cPromptV2 = TCanvas("cPromptV2", "prompt v2 versus pt")
+cPromptV2.SetCanvasSize(800, 800)
+cPromptAndNonPromptV2 = TCanvas(
+    "cPromptAndNonPromptV2", "prompt and non-prompt v2 versus pt"
+)
+cPromptAndNonPromptV2.SetCanvasSize(800, 800)
+
+leg = TLegend(0.55, 0.75, 0.88, 0.89)
+leg.SetTextSize(0.045)
+leg.SetBorderSize(0)
+leg.SetFillStyle(0)
+
+outFile = TFile(args.outFileName, "recreate")
+hNpV2vsPt = hV2[0].Clone("hNpV2vsPt")
+hPrV2vsPt = hV2[0].Clone("hPrV2vsPt")
+
+for iPt in range(hNpFrac[0].GetNbinsX()):
+
+    hGraphV2.append(TGraphErrors(-1))
+    hFracvsV2.append(TH1D(f"hFracvsV2{iPt}", "", 1000, 0.0, 1.0))
+    hFracvsV2[-1].SetDirectory(0)
+
+    TotalV2Xl = Double_t()
+    TotalV2Xh = Double_t()
+    for i in range(4):
+        TotalV2Xl += AverageV2[i].GetErrorXlow(iPt)
+        TotalV2Xh += AverageV2[i].GetErrorXhigh(iPt)
+
+    FinalErroXl.append(TotalV2Xl / 4)
+    FinalErroXh.append(TotalV2Xh / 4)
+
+gV2Np = TGraphAsymmErrors("gNpV2vsPt")
+gV2Pr = TGraphAsymmErrors("gPrV2vsPt")
+
+t = TLatex(8, 8, "")
+t.SetNDC()
+t.SetTextFont(42)
+t.SetTextSize(0.047)
+t.SetTextColor(kBlack)
+
+for iPt in range(hNpFrac[0].GetNbinsX()):
+
+    ptMin = hNpFrac[0].GetBinLowEdge(iPt + 1)
+    ptMax = ptMin + hNpFrac[0].GetBinWidth(iPt + 1)
+    ptCent = hNpFrac[0].GetBinCenter(iPt + 1)
+
+    listV2 = [hV2Tmp.GetBinContent(iPt + 1) for hV2Tmp in hV2]
+    listV2Unc = [hV2Tmp.GetBinError(iPt + 1) for hV2Tmp in hV2]
+    listNpFrac = [hNpFracTmp.GetBinContent(iPt + 1) for hNpFracTmp in hNpFrac]
+    listNpFracUnc = [hNpFracTmp.GetBinError(iPt + 1) for hNpFracTmp in hNpFrac]
+
+    for ipoint, (v2, npfrac, v2unc, npfracunc) in enumerate(
+        zip(listV2, listNpFrac, listV2Unc, listNpFracUnc)
+    ):
+
+        if iPt == 0 and ipoint == 3:  # only 3 points for the first pt bin
```

Comment:
This should be treated case by case actually, I would remove this or at least I would add a warning

Reply:
I have already re-implemented the whole code and removed this part. In run3 and for prompt, the statistic should be enough. if there is any other issue, let's fix it in the next